test/testMethods.ts

```diff
@@ -225,12 +225,18 @@ module TestMethods {
     target.node().dispatchEvent(e);
   }
 
-  export function triggerFakeKeyboardEvent(type: string, target: d3.Selection<void>, keyCode: number) {
+  export function triggerFakeKeyboardEvent(type: string, target: d3.Selection<void>, keyCode: number, options?: {[key: string]: any}) {
     let event = <KeyboardEvent> document.createEvent("Events");
     event.initEvent(type, true, true);
     event.keyCode = keyCode;
+    if (options != null ) {
+      for (var key in options) {
+        if (options.hasOwnProperty(key)) {
```

Comment:
Using `Object.keys()` will allow you to avoid having to do this check yourself:
```Typescript
Object.keys(options).forEach((key) => (<any> event)[key] = options[key] );
```